aiohttp/web_response.py

```diff
@@ -364,32 +364,52 @@ async def _prepare_headers(self) -> None:
         if self._compression:
             await self._start_compression(request)
 
+        # The following should not have a body per
+        # https://datatracker.ietf.org/doc/html/rfc9112#section-6.3
+        # status code: 204, 304, 1xx
+        # method: HEAD
+        request_indicates_empty_body = (
```

Comment:
Should we be doing this on a CONNECT 2xx response as well?

Reply:
I think from my reading, Transfer-Encoding is forbidden, and Content-Length will be the same as 304 (so, if we're removing it for 304 we should remove it here too).

Reply:
I think that works but I don't have a good test case for this

04e6cfaf80c862ed7c805e61ffb60b88e6ad3d9f